### ISSUE TITLE
testObjectLocking2 is expected test failure: Locking problem on Mac/Linux

### ISSUE THREAD

**Issue by MarcusDenker** [ApptiveGrid/MoniBase#10]
testObjectLocking2 has a comment:

```
	"Test if object locking works with transactions running in two database connections."

	"This test fails on unix based systems as locking does not rely solely on file locking but
	on the database connection. If there are two connections it does not work"
```

**Comment by noha** [comment]
This is due to the old omnibase code relying on mandatory file locking in windows. In unix/linux a flock does not lock within the same process. A process is supposed to deal best with concurrency. flock protects concurrent access by separate processes

**Comment by noha** [comment]
Fixed by #49
